### ISSUE TITLE
markup_sell_pct/markdown_buy_pct does not follow commandline args in Live trading

### ISSUE THREAD

**Issue by yang2lalang** [DeviaVir/zenbot#1150]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/zenbot

If you open a GitHub issue, here is our policy:

1. It must be a bug or a feature request.
2. The form below must be filled out.

**Here's why we have that policy**: Zenbot developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies hundreds of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using zenbot vanilla)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **Zenbot version**:  version 4 master pulled 16/01/2018
- **NodeJS version**:   v9.2.1
- **Python version**: Python 2.7.12 

- **Exact command to reproduce**:        ./zenbot.sh trade  gdax.LTC-EUR --avgpoints   1000  --avgpoints2   100  --buy_pct   100   15  --days   3  --lastpoints   100   --lastpoints2   10     --max_sell_loss_pct   5    --max_slippage_pct   5   --order_type    maker   --period  5s     --period_length    5s  --profit_stop_enable_pct   1 --profit_stop_pct   0.05 --rsi_periods   14 --sell_pct   100 --sell_stop_pct   40  --strategy    trendline  --show_options --poll_trades=1800 --markdown_buy_pct 0  --markup_sell_pct 0

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in Zenbot or a feature request.

The bug occurs in Live trading on gdax LTC/EUR, i notice the robot is not able to buy or sell as quickly due to **markdown_buy_pct** and **markup_sell_pct** . These have been explicitly set to zero but when i tap the options key "O" in live trade, i see that zenbot is using a higher value and therefore sets desired price away from the market and so the order doesnt get filled in time and the market moves away from it.

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

2018-01-16 00:57:40  164.970 LTC-EUR   -0.01%        9   59 ++    1.01351987 0.99130549 1.48066721 4.72812880 165.99809999 165.43952000 1.87494910     sell  0.00000000 LTC  20.71 EUR   +0.00%   -4.75%
2018-01-16 00:57:55  164.970 LTC-EUR   +0.00%        2   59 ++    1.01353036 0.99156645 1.46426690 4.71144758 165.96379999 165.42650000 1.86517093     sell  0.00000000 LTC  20.71 EUR   +0.00%   -4.75%
2018-01-16 00:58:05  164.970 LTC-EUR   +0.00%        0   59 ++    1.01353305 0.99182755 1.44669680 4.69467098 165.92940000 165.41348000 1.85500891     sell  0.00000000 LTC  20.71 EUR   +0.00%   -4.75%
2018-01-16 00:58:10  164.970 LTC-EUR   +0.00%        3   59 ++    1.01353195 0.99208938 1.42825802 4.68268213 165.89509999 165.40243000 1.84601234     sell  0.00000000 LTC  20.71 EUR   +0.00%   -4.75%
2018-01-16 00:58:40  164.980 LTC-EUR   +0.00%       20   59 ++    1.01353555 0.99235135 1.40856256 4.66559963 165.86070000 165.38936000 1.83511181     sell  0.00000000 LTC  20.71 EUR   +0.00%   -4.76%
2018-01-16 00:58:45  164.970 LTC-EUR   -0.01%        2   59 ++    1.01353927 0.99261345 1.38785977 4.64570139 165.82650000 165.37534000 1.82306057     sell  0.00000000 LTC  20.71 EUR   +0.00%   -4.75%
2018-01-16 00:58:55  164.950 LTC-EUR   -0.02%        0   58 ++    1.01353878 0.99287510 1.36604727 4.62570537 165.79220000 165.36132000 1.81064181     sell  0.00000000 LTC  20.71 EUR   +0.00%   -4.74%
2018-01-16 00:59:05  164.050 LTC-EUR   -0.55%        1    -- 38   1.01354689 0.99313749 1.34292674 4.60250963 165.75760000 165.34628000 1.79686654     sell  0.00000000 LTC  20.71 EUR   +0.00%   -4.22%
2018-01-16 00:59:15  164.930 LTC-EUR   +0.53%       13   55 ++    1.01355488 0.99339941 1.32698763 4.58296357 165.71410000 165.33153000 1.78637665     sell  0.00000000 LTC  20.71 EUR   +0.00%   -4.73%
2018-01-16 00:59:35  164.910 LTC-EUR   -0.02%        0   54 +     1.01356080 0.99366147 1.30121270 4.56694042 165.67930000 165.31902000 1.77394094     sell  0.00000000 LTC  20.71 EUR   +0.00%   -4.72%
2018-01-16 00:59:42  164.890 LTC-EUR   -0.02%       16   54 +     1.01356446 0.99386679 1.27429691 4.54851034 165.64440000 165.30565000 1.76043628           0.00000000 LTC  20.71 EUR   +0.00%   -4.71%
{
    avgpoints: 1000,
    avgpoints2: 100,
    buy_pct: 100,
    days: 3,
    lastpoints: 100,
    lastpoints2: 10,
    max_sell_loss_pct: '5',
    max_slippage_pct: '5',
    order_type: 'maker',
    period: '5s',
    period_length: '5s',
    profit_stop_enable_pct: 1,
    profit_stop_pct: 0.05,
    rsi_periods: 14,
    sell_pct: 100,
    sell_stop_pct: 40,
    strategy: 'trendline',
    show_options: true,
    poll_trades: 1800,
    buy_stop_pct: 0,
    order_adjust_time: 5000,
    order_poll_time: 3000,
    markdown_buy_pct: 1.7604362810190213,
    markup_sell_pct: 1.7604362810190213,
    currency_capital: 1000,
    asset_capital: 0,
    avg_slippage_pct: 0.045,
    currency_increment: null,
    keep_lookback_periods: undefined,
    debug: undefined,
    stats: true,
    mode: 'live',
    selector: {
        exchange_id: 'gdax',
        product_id: 'LTC-EUR',
        asset: 'LTC',
        currency: 'EUR',
        normalized: 'gdax.LTC-EUR'
    },
    min_periods: 15000
}
2018-01-16 00:59:45  164.890 LTC-EUR   -0.02%       16   54 +     1.01356446 0.99386679 1.27429691 4.54851034 165.64440000 165.30565000 1.76043628     sell  0.00000000 LTC  20.71 EUR   +0.00%   -4.71%
2018-01-16 01:00:00  164.900 LTC-EUR   +0.00%        0   54 +     1.01356812 0.99407220 1.24571239 4.52473877 165.60920000 165.29055000 1.74482276     sell  0.00000000 LTC  20.71 EUR   +0.00%   -4.71%
2018-01-16 01:00:06  164.900 LTC-EUR   +0.00%        1   54 +     1.01357788 0.99427828 1.21561439 4.50391670 165.57420000 165.27645000 1.72963089           0.00000000 LTC  20.71 EUR   +0.00%   -4.71%

**Comment by abduegal** [comment]
Those are strange values for the markup and markdown. Could you verify that these values have not been set in the conf.js and conf-sample.js
The value: 1.7604362810190213 does not exist in the zenbot code, so they must be set from somewhere else.

**Comment by yang2lalang** [comment]
@abduegal  i have checked the two files and in both files markup and markdown are set to zero

i have  attached some screenshot

Does anyone who uses gdax have this same issue in 
![confjs](https://user-images.githubusercontent.com/12873450/34987417-29763a42-fa5f-11e7-90b7-bf315ae7e83b.png)
![confsamplejs](https://user-images.githubusercontent.com/12873450/34987421-2d8c2e16-fa5f-11e7-927b-2ff470df8a7e.png)

Live trading

**Comment by abduegal** [comment]
allright I found it:
I reproduced the problem and it seems to happen in the trendline strategy code. The markdown is recalculated there as part of the strategy:

        s.stdevs = stats.stdev(tls)
        s.stdevl = stats.stdev(tll)
        s.means = math.mean(tls)
        s.meanl = math.mean(tll)
        s.pcts = s.stdevs / s.means
        s.pctl = s.stdevl / s.meanl
        s.options.markup_sell_pct = math.mean(s.pcts, s.pctl) * 100
        s.options.markdown_buy_pct = math.mean(s.pcts, s.pctl) * 100

You can find this in "extensions/strategies/trendline/strategy.js"
I'm not sure if it is a bug or if it is a part of the strategy.

**Comment by yang2lalang** [comment]
@abduegal  yes i just saw it. i will test it more to see if it is better set to 0 or not. it is possible this markup does not enable efficient buying in Live trading

i will report the results here

**Comment by yang2lalang** [comment]
i can confirm that making the changes

          //s.options.markup_sell_pct = math.mean(s.pcts, s.pctl) * 100
          //s.options.markdown_buy_pct = math.mean(s.pcts, s.pctl) * 100
          s.options.markup_sell_pct = 0
          s.options.markdown_buy_pct = 0

causes zenbot to buy as soon as the signal is issued making for a more efficient buy or sell

i cannot say though if this is a delibrate design to markup or markdown in trendline strategy

I think the strategy designer can respond to this

For mei am satisfied with this solution
